src/server/session.ts

```diff
@@ -2198,7 +2198,22 @@ namespace ts.server {
             const scriptInfo = project.getScriptInfoForNormalizedPath(file)!;
             const { startPosition, endPosition } = this.getStartAndEndPosition(args, scriptInfo);
 
-            const codeActions = project.getLanguageService().getCodeFixesAtPosition(file, startPosition, endPosition, args.errorCodes, this.getFormatOptions(file), this.getPreferences(file));
+            let codeActions: readonly CodeFixAction[];
+            try {
+                codeActions = project.getLanguageService().getCodeFixesAtPosition(file, startPosition, endPosition, args.errorCodes, this.getFormatOptions(file), this.getPreferences(file));
+            }
+            catch(e) {
+                const ls = project.getLanguageService();
+                const existingDiagCodes =
+                    flatMap(["getSyntacticDiagnostics", "getSemanticDiagnostics", "getSuggestionDiagnostics"] as const,
+                            getter => ls[getter](file))

```

Comment:
I won’t block on code style, but feel I should point out that this indirection is _more characters_ than just spreading the result of three function calls into an array...

```
flatMap(["getSyntacticDiagnostics", "getSemanticDiagnostics", "getSuggestionDiagnostics"] as const, getter => ls[getter](file)).map(...)
[...ls.getSyntacticDiagnostics(file), ...ls.getSemanticDiagnostics(file), ...ls.getSuggestionDiagnostics(file)].map(...)
```

Reply:
(Switched this.)